src/main/java/htsjdk/samtools/util/IntervalList.java

```diff
@@ -869,4 +844,68 @@ public int hashCode() {
         result = 31 * result + intervals.hashCode();
         return result;
     }
+
+    public static class IntervalMerger implements Iterable<Interval> {
+
+        Iterator<Interval> inputIntervals;
+        Interval current = null;
+        final boolean combineAbuttingIntervals;
+        final boolean enforceSameStrands;
+        final boolean concatenateNames;
+        final List<Interval> toBeMerged = new ArrayList<>();
+
+        public IntervalMerger(Iterator<Interval> intervals, final boolean combineAbuttingIntervals, final boolean enforceSameStrand, final boolean concatenateNames) {
```

Comment:
This should probably validate that the input is sorted.  At the very least it needs javadoc specifying that sorting is required.

Reply:
good point.